packages/react/src/components/button/button.tsx

```diff
@@ -10,7 +10,7 @@ export const Button: ButtonComponent = ({
 }: ButtonProps<'a'> | ButtonProps<'button'>) => {
   const { disabled } = useField();
 
-  const Element = 'href' in restProps ? 'a' : 'button';
+  const Element = (restProps as ButtonProps<'a'>).href ? 'a' : 'button';
```

Comment:
Storybook passing `href: undefined` meant this rendered as `<a>`.

This is a fix not for our end customers, but for Storybook and ourselves.

Reply:
I just checked on UI and don't get this, how did you find this out?

Reply:
I didn't, our automated visual regression tests did 🙊 

https://onfido.github.io/castor/iframe.html?id=react-button--all-combinations&viewMode=story

Reply:
Same thing for me
![image](https://user-images.githubusercontent.com/18623773/114403534-ca874c00-9b9c-11eb-95f0-2f66e09f11f5.png)

Reply:
I feel like this is some sort of Storybook bug, try this and scroll down: https://onfido.github.io/castor/?path=/docs/react-button--playground

Reply:
That works, so it seems that if the entrypoint story (the one on the url) is "All Combinations" it passes `{ href: undefined }`, otherwise it passes `{ }`.

Not sure if a bug, but definitely different behaviours depending on path.

Reply:
Regardless, checking for value in the Button is probably better than checking for key, as we want this to render as `<button>`:

```tsx
<Button href={undefined} />
```

Reply:
Right, so anyways your change seems fine. Resolving this now.